search/indexing_api.py

```diff
@@ -193,13 +193,12 @@ def serialize_program_enrolled_user(program_enrollment):
 def filter_current_work(profile):
     """
     Remove work_history objects that are not current
+    Args:
+        profile: serialized profile
+    Returns:
+        profile: profile with filtered current work_history list
     """
-    current_employment = []
-    for work in profile['work_history']:
-        if not work['end_date']:
-            current_employment.append(work)
-
-    profile['work_history'] = current_employment
+    profile['work_history'] = [work for work in profile['work_history'] if not work['end_date']]
```

Comment:
I forgot to mention this earlier but we should make a defensive copy of `profile` so that the original is not modified. Can you add a `profile = dict(profile)` before this line?